recipes/autoconf/all/conanfile.py

```diff
@@ -1,122 +1,146 @@
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        datarootdir = self._datarootdir
-        prefix = self.package_folder
         if self.settings.os == "Windows":
-            datarootdir = tools.unix_path(datarootdir)
-            prefix = tools.unix_path(prefix)
-        conf_args = [
-            "--datarootdir={}".format(datarootdir),
-            "--prefix={}".format(prefix),
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    def _patch_files(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        with tools.environment_append(tools.RunEnvironment(self).vars):
-            yield
+            if is_msvc(self):
+                build = "{}-{}-{}".format(
+                    "x86_64" if self._settings_build.arch == "x86_64" else "i686",
+                    "pc" if self._settings_build.arch == "x86" else "win64",
+                    "mingw32")
+                host = "{}-{}-{}".format(
+                    "x86_64" if self.settings.arch == "x86_64" else "i686",
+                    "pc" if self.settings.arch == "x86" else "win64",
+                    "mingw32")
+                tc.configure_args.append(f"--build={build}")
+                tc.configure_args.append(f"--host={host}")
+
+        env = tc.environment()
+        env.define("INSTALL", unix_path(self, str(self.source_path.joinpath('build-aux', 'install-sh'))))
+        tc.generate(env)
+
+        deps = AutotoolsDeps(self)
+        deps.generate()
+
+        ms = VirtualBuildEnv(self)
+        ms.generate(scope="build")
 
     def build(self):
-        self._patch_files()
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        replace_in_file(self, self.source_path.joinpath("Makefile.in"), "M4 = /usr/bin/env m4", "#M4 = /usr/bin/env m4")
+
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "info"))
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        autotools = Autotools(self)
+        autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])  # Need to specify the `DESTDIR` as a Unix path, aware of the subsystem
+
+        copy(self, "COPYING*", src=self.source_folder, dst=self.package_path.joinpath("licenses"))
+        rmdir(self, self.package_path.joinpath("res", "info"))
```

Comment:
```suggestion
        rmdir(self, self.package_path.joinpath("res", "info"))
```
Use self.package_folder only for now. `package_path` is not documented in Conan docs, which means, it should be considered as private.

Reply:
@jellespijker I opened the issue https://github.com/conan-io/docs/issues/2753 asking to document it.

Reply:
I often use all the `_path` synonyms instead of the folder. Since it makes for shorter calls and gives access to `glob()` and `exists()` saving me an `import os`. I also think it makes things more readable.

I appreciated the PR :thumbs_up:

Reply:
Yes, let's wait and see if Conan team accepts. I gonna ask them next week, to avoid delays, but it seems okay, I don't think they will change it in the future.

Reply:
Any word if this is accepted?